domain/secretbackend/service/service.go

```diff
@@ -700,9 +700,53 @@ func (s *Service) RotateBackendToken(ctx context.Context, backendID string) erro
 	return errors.Trace(err)
 }
 
-// SetModelSecretBackend sets the secret backend for the given model UUID.
+// GetModelSecretBackend returns the secret backend name for the given model UUID, returning an error
+// satisfying [modelerrors.NotFound] if the model provided does not exist.
+func (s *Service) GetModelSecretBackend(ctx context.Context, modelUUID coremodel.UUID) (string, error) {
+	backendName, modelType, err := s.st.GetModelSecretBackend(ctx, modelUUID)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	switch modelType {
+	case coremodel.IAAS:
+		if backendName == provider.Internal {
```

Comment:
Is this actually valid logic. Why hide what the secret backend is and or not allow a user to say use Kubernetes of it is a Kubernetes model. Seems like the easier approach to just report what it is and get rid of a possible error case.

Reply:
Yes. This is the current logic.